rosdep/base.yaml

```
@@ -2919,6 +2919,9 @@ libopencv-dev:
   openembedded: [opencv@meta-oe]
   rhel: [opencv-devel]
   ubuntu: [libopencv-dev]
+libopencv-imgproc-dev:
+  debian: [libopencv-imgproc-dev]
+  ubuntu: [libopencv-imgproc-dev]
```

* Arch: https://www.archlinux.org/packages/extra/x86_64/opencv/
* Fedora: https://apps.fedoraproject.org/packages/opencv-devel
* Gentoo: https://packages.gentoo.org/packages/media-libs/opencv
* RHEL: [opencv-devel-2.4.5-3.el7.x86_64.rpm](http://mirror.centos.org/centos/7.8.2003/os/x86_64/Packages/)

```suggestion
  arch: [opencv]
  debian: [libopencv-imgproc-dev]
  fedora: [opencv-devel]
  gentoo: [media-libs/opencv]
  rhel: [opencv-devel]
  ubuntu: [libopencv-imgproc-dev]
```

---

Thank you for your suggestion @cottsay. 
I'm wondering wouldn't it create some confusion to have `libopencv-imgproc-dev` resolving to the same thing than `libopencv-dev` for arch fedora gentoo rhel ?
On the other hand, I understand the benefit of having a package relying on `libopencv-imgproc-dev` running seamlessly on all those distros.
Is it a common practice for ROS dependencies?

---

This is common practice across rosdep. Another common example is Boost, where Ubuntu creates individual `-dev` packages for each component and Fedora/RHEL have a single `boost-devel` package that contains all of the components.
In build-from-source platforms, there aren't subpackages at all like there are in Debian, Ubuntu, Fedora, and RHEL. That's why you don't see `-dev` packages for Arch and Gentoo rules.
You can think of it like this: On Ubuntu, `libopencv-imgproc-dev` provides a "capability", the headers and libraries for developing against `libopencv_imgproc.so`. In these other platforms, the package maintainers didn't think it was necessary to create separate subpackages for this capability, so they included it with another (larger) package. The capability is there, you just get some extra stuff along with it.

---

@cottsay Thank you for this very clear explanation. I committed your suggestion.